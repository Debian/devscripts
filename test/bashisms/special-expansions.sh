#!/bin/sh

set -- foo bar moo

echo BASHISM: ${#@}
echo BASHISM: ${#*}

echo BASHISM: ${@%f*}
echo BASHISM: ${*%f*}
echo BASHISM: ${@%%f*}
echo BASHISM: ${*%%f*}

echo BASHISM: ${@#*o}
echo BASHISM: ${*#*o}
echo BASHISM: ${@##*o}
echo BASHISM: ${*##*o}

echo BASHISM: ${@/?/u}
echo BASHISM: ${*/?/u}
echo BASHISM: ${@/?/}
echo BASHISM: ${*/?/}

echo BASHISM: ${@:2}
echo BASHISM: ${*:2}
echo BASHISM: ${@:1:1}
echo BASHISM: ${*:1:1}
