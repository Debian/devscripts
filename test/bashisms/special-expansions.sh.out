possible bashism in bashisms/special-expansions.sh line 5 (${#@} or ${#*}):
echo BASHISM: ${#@}
possible bashism in bashisms/special-expansions.sh line 6 (${#@} or ${#*}):
echo BASHISM: ${#*}
possible bashism in bashisms/special-expansions.sh line 8 (${[@|*]#[#]pat} or ${[@|*]%[%]pat}):
echo BASHISM: ${@%f*}
possible bashism in bashisms/special-expansions.sh line 9 (${[@|*]#[#]pat} or ${[@|*]%[%]pat}):
echo BASHISM: ${*%f*}
possible bashism in bashisms/special-expansions.sh line 10 (${[@|*]#[#]pat} or ${[@|*]%[%]pat}):
echo BASHISM: ${@%%f*}
possible bashism in bashisms/special-expansions.sh line 11 (${[@|*]#[#]pat} or ${[@|*]%[%]pat}):
echo BASHISM: ${*%%f*}
possible bashism in bashisms/special-expansions.sh line 13 (${[@|*]#[#]pat} or ${[@|*]%[%]pat}):
echo BASHISM: ${@#*o}
possible bashism in bashisms/special-expansions.sh line 14 (${[@|*]#[#]pat} or ${[@|*]%[%]pat}):
echo BASHISM: ${*#*o}
possible bashism in bashisms/special-expansions.sh line 15 (${[@|*]#[#]pat} or ${[@|*]%[%]pat}):
echo BASHISM: ${@##*o}
possible bashism in bashisms/special-expansions.sh line 16 (${[@|*]#[#]pat} or ${[@|*]%[%]pat}):
echo BASHISM: ${*##*o}
possible bashism in bashisms/special-expansions.sh line 18 (${parm/?/pat[/str]}):
echo BASHISM: ${@/?/u}
possible bashism in bashisms/special-expansions.sh line 19 (${parm/?/pat[/str]}):
echo BASHISM: ${*/?/u}
possible bashism in bashisms/special-expansions.sh line 20 (${parm/?/pat[/str]}):
echo BASHISM: ${@/?/}
possible bashism in bashisms/special-expansions.sh line 21 (${parm/?/pat[/str]}):
echo BASHISM: ${*/?/}
possible bashism in bashisms/special-expansions.sh line 23 (${foo:3[:1]}):
echo BASHISM: ${@:2}
possible bashism in bashisms/special-expansions.sh line 24 (${foo:3[:1]}):
echo BASHISM: ${*:2}
possible bashism in bashisms/special-expansions.sh line 25 (${foo:3[:1]}):
echo BASHISM: ${@:1:1}
possible bashism in bashisms/special-expansions.sh line 26 (${foo:3[:1]}):
echo BASHISM: ${*:1:1}
